src/System.Device.Gpio/System/Device/Gpio/Drivers/RaspberryPi3Driver.cs

```diff
@@ -23,6 +23,41 @@ public class RaspberryPi3Driver : GpioDriver
         private readonly Set_Register _setClearRegister;
         private readonly Get_Register _getClearRegister;
 
+        /// <summary>
+        /// Default Gpio mode. Use <see cref="SetPinMode"/> to switch between input and output.
+        /// </summary>
+        public static readonly ExtendedPinMode GpioMode = new RaspberryPiExtendedPinMode("GPIO", 0);
+
+        /// <summary>
+        /// Alternate mode 0 for Raspberry Pi
+        /// </summary>
+        public static readonly ExtendedPinMode Alt0Mode = new RaspberryPiExtendedPinMode("Alt0", 0b100);
+
+        /// <summary>
+        /// Alternate mode 1 for Raspberry Pi
+        /// </summary>
+        public static readonly ExtendedPinMode Alt1Mode = new RaspberryPiExtendedPinMode("Alt1", 0b101);
+
+        /// <summary>
+        /// Alternate mode 2 for Raspberry Pi
+        /// </summary>
+        public static readonly ExtendedPinMode Alt2Mode = new RaspberryPiExtendedPinMode("Alt2", 0b110);
+
+        /// <summary>
+        /// Alternate mode 3 for Raspberry Pi
+        /// </summary>
+        public static readonly ExtendedPinMode Alt3Mode = new RaspberryPiExtendedPinMode("Alt3", 0b111);
+
+        /// <summary>
+        /// Alternate mode 4 for Raspberry Pi
+        /// </summary>
+        public static readonly ExtendedPinMode Alt4Mode = new RaspberryPiExtendedPinMode("Alt4", 0b011);
+
+        /// <summary>
+        /// Alternate mode 5 for Raspberry Pi
+        /// </summary>
+        public static readonly ExtendedPinMode Alt5Mode = new RaspberryPiExtendedPinMode("Alt5", 0b010);

```

Comment:
Forgive me if I got here just now. Just for the knowledge this should not be done or decided in user space, it is not a good practice (at least in Linux). This should be decided in the description of the hardware in the Device Tree Source or Device tree overlay.

Don't get me wrong, nothing against its implementation.

Reply:
Sorry for being annoying with everything that touches what the Linux Kernel already does. I just have a concern that dotnet/iot becomes a library that is not a first class citizen when using Linux resources. We also have to be careful about how to recommend this use. For use in development environment it may be ok ...

Reply:
@microhobby I'm sorry, but I don't understand. Where does the linux kernel already provide this information or these functions? The latest libgpiod version for the raspi is still 1.2. Of course if that driver supports these features, that can be extended as well.

Reply:
@microhobby the only reason we consider APIs like this are issues like: https://github.com/dotnet/iot/issues/874 - kernel currently doesn't seem to be solving this problem correctly, how would you recommend to do it otherwise?

Reply:
@pgrawehr the alternative modes are set by the Kernel on the boot, the pinctrl sub system handle this. The pinctrl subsystem load these information from the hardware description, the device tree source. All boards have it own device tree source: https://git.kernel.org/pub/scm/linux/kernel/git/torvalds/linux.git/tree/arch/arm/boot/dts

gpiolib and libgpiod do not change alternative functions, were not designed for that. They were designed only to change pin states that are in the alternative GPIO function. Even because changing the alternate function of a pin is not a user space taks, as I said it happens in kernel space at boot by the pinctrl driver.

Reply:
@krwq in fact I think it’s not that the kernel isn’t solving the problem correctly, actually we’re overwriting what the kernel knows about the hardware. 

Since for RPI it seems we are using `/dev/mem` anyway I don't see why not follow with this solution, we should only be careful when recommending this. For development and testing environments I'm not seeing problems.

This is my humble opinion, let me know what you guys think.